### ISSUE TITLE
Docker Build failed with exit code 255

### ISSUE THREAD

**Issue by eclipses** [microsoft/navcontainerhelper#2360]
Hi since yesterday the creation of containers started failing with the following error:
"Docker Build failed with exit code 255"
I verified and there are not pending Windows Updates in the VM.

**Scripts used to create container and cause the issue**
```
    New-BCContainer -accept_eula -includeAL -accept_outdated -imageName ci -licenseFile "C:\Users\Developer\Documents\Licenses\license.flf" `
                    -doNotCheckHealth `
                    -updateHosts `
                    -containerName dev `
                    -artifactUrl  "https://bcartifacts.azureedge.net/sandbox/19.5.36567.37901/base"`
                    -auth "UserPassword" `
                    -Credential $credential `
                    -additionalParameters $additionalParameters `
                    -includeTestToolkit `
                    -includeTestFrameworkOnly `
                    -doNotUseRuntimePackages `
                    -enableTaskScheduler:$false
```

**Full output of scripts**
```

Select Version (current, nextminor, nextmajor) (default current): 
Enter name of container (enter for dev): 
Set bcContainerHelperVersion = latest
##vso[task.setvariable variable=bcContainerHelperVersion]latest
Version: latest
BcContainerHelper 3.0.3 is installed
Determine latest BcContainerHelper version
BcContainerHelper 3.0.3 is the latest version
BcContainerHelper is version 3.0.3
BcContainerHelper is running as administrator
Host is Microsoft Windows 11 Enterprise - 21H2
Docker Client Version is 20.10.13
Docker Server Version is 20.10.13
Fetching all docker images
Fetching all docker volumes
ArtifactUrl and ImageName specified
WARNING: Unable to find matching generic image for your host OS. Using mcr.microsoft.com/businesscentral:10.0.20348.587
Building multitenant image ci:sandbox-19.5.36567.37901-base-mt based on mcr.microsoft.com/businesscentral:10.0.20348.587 with https://bcartifacts.azureedge.net/sandbox/19.5.36567.37901/base
Pulling latest image mcr.microsoft.com/businesscentral:10.0.20348.587
10.0.20348.587: Pulling from businesscentral
Digest: sha256:c9ebf4f5a6d5d817aa883790f8eaac4a463c1412fc7be73573a81d7858c2e999
Status: Image is up to date for mcr.microsoft.com/businesscentral:10.0.20348.587
mcr.microsoft.com/businesscentral:10.0.20348.587
Generic Tag: 1.0.2.1Container OS Version: 10.0.20348.587 (ltsc2022)
Host OS Version: 10.0.22000.556 (21H2)
Using process isolation
Using license file C:\Users\Developer\Documents\Licenses\license.flf
Files in c:\bcartifacts.cache\0phymkw2.tfp\my:
Copying Platform Artifacts
c:\bcartifacts.cache\sandbox\19.5.36567.37901\platform
Copying Database
Copying Licensefile
c:\bcartifacts.cache\0phymkw2.tfp
Sending build context to Docker daemon  1.579GB

Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.20348.587
 ---> 7c9b57e1dc38
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/19.5.36567.37901/base filesOnly=False
 ---> Using cache
 ---> 97f80fdde964
Step 3/6 : COPY my /run/
 ---> Using cache
 ---> 05f8031fbf97
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> 29d4989aa171
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestFrameworkOnly
 ---> Running in 467e9505f49c
Using installer from C:\Run\150-new
Installing Business Central
Installing from DVD
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
C:\NAVDVD\ServiceTier\Program Files
C:\NAVDVD\ServiceTier\System64Folder
Copying PowerShell Scripts
C:\NAVDVD\WindowsPowerShellScripts\Cloud\NAVAdministration
C:\NAVDVD\WindowsPowerShellScripts\WebSearch
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Determining Database Collation
The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant -includeTestToolkit -includeTestFrameworkOnly' returned a non-zero code: 255
New-Bcimage Telemetry Correlation Id: a6e3ee7d-5a4d-4d94-bffe-7089d14282a9
New-BCContainer Telemetry Correlation Id: a2fb881d-946c-45e1-a45a-f76ed51a97ad
Docker Build failed with exit code 255
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.3\ContainerHandling\New-NavImage.ps1:635 char:25
+ ...              throw "Docker Build failed with exit code $LastExitCode"
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Docker Build failed with exit code 255:String) [], RuntimeException
    + FullyQualifiedErrorId : Docker Build failed with exit code 255
```

**Comment by TheDoubleH** [comment]
I think this may be related to two issues I am having as well.

A recent Windows Update AND a recent Docker Desktop update.

`New-BcContainerWizard` tell me that I will now be suggested to use Hyperv isolation. However, when executing the script it is running as Process Isolation

```
BcContainerHelper is version 3.0.3
BcContainerHelper is running as administrator
Host is Microsoft Windows 11 Pro - 21H2
Docker Client Version is 20.10.13
Docker Server Version is 20.10.13
```

and

```
Container OS Version: 10.0.20348.587 (ltsc2022)
Host OS Version: 10.0.22000.556 (21H2)
Using process isolation
```

adding `-isolation 'hyperv'` changes nothing

**Comment by TheDoubleH** [comment]
**Update**

After much crying and re-thinking my life-choices, I was able to figure out (with the help from https://github.com/microsoft/navcontainerhelper/issues/2358) that adding `$isolation = "hyperv"` to the `New-NavImage` just before this line

https://github.com/microsoft/navcontainerhelper/blob/4719802eb00b1b6babe398dcb27d26929f808915/ContainerHandling/New-NavImage.ps1#L413

resolved the issue.

So all in all - The issue is that currently `$isolation` is set to `process` regardless if I set it to `hyperv`

**Comment by freddydk** [comment]
When you specify -isolation hyperv to New-BcContainer, then this should also take effect when building an image and I can see that it doesn't always...
@eclipses  with the next preview, you can try to add -isolation hyperv

**Comment by eclipses** [comment]
Hi Freddy, 

Using hyperv isolation and the preview version of BcContainerHelper it worked.

Thank you.

You can close this issue when releasing the new version.

**Comment by freddydk** [comment]
fixed in 3.0.4
